src/cpp/rtps/participant/RTPSParticipantImpl.cpp

```diff
@@ -2160,6 +2167,10 @@ void RTPSParticipantImpl::deleteAllUserEndpoints()
         {
             static_cast<BaseReader*>(endpoint)->local_actions_on_reader_removed();
         }
+        else if (WRITER == kind)
+        {
+            dynamic_cast<BaseWriter*>(endpoint)->local_actions_on_writer_removed();
```

Comment:
We know that endpoint would be a `BaseWriter`, so a `static_cast` would be enough here

Reply:
Apply in c65c3e9